src/base.jl

```diff
@@ -176,6 +176,21 @@ Base.length(prob::Problem) = length(prob.solver)
 
 Base.eltype{O,S}(::Type{Problem{O,S}}) = eltype(O)
 
+if VERSION >= v"0.5.0-rc0"
+    """
+    Makes some generic operations on iterators work, like
+    generator comprehensions:
+        tgen=(t for (t,y) in sol)
+        tout=collect(tgen)
+    or
+        errgen=(y-[exp(t)] for (t,y) in sol)
+        errout=collect(errgen)
+
+    TODO: doesn't work for 0.4 and might have show issues due to non-copying output
```

Comment:
I'd be fine with dropping 0.4 compatibility.  But then maybe we shouldn't as other packages will take some time to adapt 0.5.